backend/services/mgmt/v1alpha1/connection-data-service/connection-data.go

```diff
@@ -404,12 +406,20 @@ func (s *Service) GetConnectionSchema(
 
 		schemas := []*mgmtv1alpha1.DatabaseColumn{}
 		for _, col := range dbschema {
+			hasDefault := false
+			if col.ColumnDefault != nil {
+				// ColumnDefault is an interface we need to check the value to see if it's set or not
+				value := reflect.ValueOf(col.ColumnDefault)
+				hasDefault = !value.IsZero()
+			}
+
```

Comment:
If you go into the `system.sql` file for postgres queries, there are two queries in there that can be updated that will make all of this code go away.

Look for this block (shows up twice):

```
    COALESCE(
        pg_catalog.pg_get_expr(d.adbin, d.adrelid),
        ''
    ) AS column_default,
```

That is the bit that calculates this column. However, sqlc isn't smart enough to know the type, so we just need to cast it to text like so:

```
    COALESCE(
        pg_catalog.pg_get_expr(d.adbin, d.adrelid),
        ''
    )::text AS column_default,
```

This will cause the Go struct to appear as a string. Then all of this reflect stuff can be removed.

Reply:
Sweet - didn't know this was there. Updated! Also had to update a few other places that relied on that column being an interface